### ISSUE TITLE
[JIT] UserWarning: `optimize` is deprecated and has no effect. Use `with torch.jit.optimized_execution() instead

### ISSUE THREAD

**Issue by merisgong208** [pytorch/pytorch#34703]
## To Reproduce

1.
1.
1.

## Expected behavior

---------------------------------------------
model = mysfv2.shufflenet_v2_x0_5()
model.load_state_dict(torch.load('shufflenetv2_x0.5-f707e7126e.pth'))
model.eval()

example = torch.randn(1,3,224,224)
traced_script_module = torch.jit.script(model, example) #warning here
traced_script_module.save('shufflenetv2_x0.5_trace.pt')
print('done')

~~~~~~~~~~~~~
·problem here:
while I try to produce a .pt for c++,  torch.jit.script() will cause a warning as title says.
And finally a 'shufflenetv2_x0.5_trace.pt' file can be produced.
But when I use it in c++, the speed is quite slow, +10s, e.g.

I also tried :

with torch.jit.optimized_execution():
       traced_script_module = torch.jit.script(model, example) 
       ...
and the problem still exists.

any way to solve it?

-----------------------------------------------------
## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.4):
 - OS (e.g., win7):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:4
 - CUDA/cuDNN version:none
 - GPU models and configuration:
 - Any other relevant information:

## Additional context

<!-- Add any other context about the problem here. -->


cc @suo

**Comment by zdevito** [comment]
optimized execution is on by default, so adding that resource guard is not expected to speed things up. One thing to do is to make sure you are not timing the first run of the model in C++. On the first run there may be a lot of initial passes running that will make the model appear slow. It is a good idea to warm up execute before timing it.

**Comment by merisgong208** [comment]
@zdevito thanks.

**Comment by sevakharutyunyan** [comment]
@zdevito Could you show an example of first warming up the cpp code before timing?

**Comment by BruceDai003** [comment]
So, what's the suggested way to remove that warning?

**Comment by BruceDai003** [comment]
> @zdevito Could you show an example of first warming up the cpp code before timing?
You can try
```
traced_script_module = torch.jit.script(model, example) # Warm up for 1 step
print(np.mean([timer(traced_script_module, example) for _ in range(100)]))
```
